docs/common/craft-parts/explanation/migration.rst

```diff
@@ -0,0 +1,115 @@
+.. _lifecycle:
+
+**************
+File migration
+**************
```

Comment:
Not covered here: file collision, overlay-to-partition migration details, partitioning details. These can be documented separately.